addons/html_editor/static/tests/link/popover.test.js

```diff
@@ -198,90 +198,206 @@ describe("Incorrect URL should be corrected", () => {
     });
 });
 
-describe("Link creation by powerbox/toolbar", () => {
-    test("click on link command in powerbox should create a link element and open the linkpopover", async () => {
-        const { editor, el } = await setupEditor("<p>ab[]</p>");
-        insertText(editor, "/link");
-        await animationFrame();
-        expect(".active .o-we-command-name").toHaveText("Link");
-
-        click(".o-we-command-name:first");
-        expect(cleanLinkArtifacts(getContent(el))).toBe("<p>ab<a>[]</a></p>");
-        await animationFrame();
-        await waitFor(".o-we-linkpopover");
-        expect(".o-we-linkpopover").toHaveCount(1);
+describe("Link creation", () => {
+    describe("Creation by space", () => {
+        test("typing valid URL + space should convert to link", async () => {
+            const { editor, el } = await setupEditor("<p>[]</p>");
+            insertText(editor, "http://google.co.in");
+            insertText(editor, " ");
+            expect(cleanLinkArtifacts(getContent(el))).toBe(
+                '<p><a href="http://google.co.in">http://google.co.in</a> []</p>'
+            );
+        });
+        test("typing invalid URL + space should not convert to link", async () => {
+            const { editor, el } = await setupEditor("<p>[]</p>");
+            insertText(editor, "www.odoo");
+            insertText(editor, " ");
+            expect(cleanLinkArtifacts(getContent(el))).toBe("<p>www.odoo []</p>");
+        });
     });
+    describe("Creation by powerbox", () => {
+        test("click on link command in powerbox should create a link element and open the linkpopover", async () => {
+            const { editor, el } = await setupEditor("<p>ab[]</p>");
+            insertText(editor, "/link");
+            await animationFrame();
+            expect(".active .o-we-command-name").toHaveText("Link");
 
-    test("when create a new link by powerbox and not input anything, the link should be removed", async () => {
-        const { editor, el } = await setupEditor("<p>ab[]</p>");
-        insertText(editor, "/link");
-        await animationFrame();
-        click(".o-we-command-name:first");
-        await animationFrame();
-        await waitFor(".o-we-linkpopover");
-        expect(".o-we-linkpopover").toHaveCount(1);
-        expect(cleanLinkArtifacts(getContent(el))).toBe("<p>ab<a>[]</a></p>");
-
-        const pNode = queryOne("p");
-        setSelection({
-            anchorNode: pNode,
-            anchorOffset: 0,
-            focusNode: pNode,
-            focusOffset: 0,
+            click(".o-we-command-name:first");
+            expect(cleanLinkArtifacts(getContent(el))).toBe("<p>ab<a>[]</a></p>");
+            await animationFrame();
+            await waitFor(".o-we-linkpopover");
```

Comment:
you don t need to wait animationFrame and waitFor. You should wait only one. 
await animationFrame() = you're waiting for a rendering owl 
await waitFor(...) = you wait for an element to be present. In general, you use it when you know that owl rendering is going to be requested in async.